maven/tests/com.liferay.ide.maven.core.tests/src/com/liferay/ide/maven/core/tests/MavenModuleProjectTests.java

```diff
@@ -589,49 +589,125 @@ public void testProjectTemplateFormField() throws Exception
+    @Test
+    public void testProjectTemplateNpmBillboardjsPortlet() throws Exception
+    {
+        NewLiferayModuleProjectOp op = NewLiferayModuleProjectOp.TYPE.instantiate();
 
-        assertTrue( project.getFile( "target/layout-test-1.0.0-SNAPSHOT.war" ).exists() );
+        op.setProjectName( "npm-billboardjs-portlet-test" );
+        op.setProjectProvider( "maven-module" );
+        op.setProjectTemplateName( "npm-billboardjs-portlet" );
+
+        createAndBuild(op);
     }
 
     @Test
-    public void testProjectTemplateMvcPortlet() throws Exception
+    public void testProjectTemplateNpmIsomorphicPortlet () throws Exception
     {
         NewLiferayModuleProjectOp op = NewLiferayModuleProjectOp.TYPE.instantiate();
 
-        op.setProjectName( "mvc-portlet-test" );
+        op.setProjectName( "npm-isomorphic-portlet-test" );
         op.setProjectProvider( "maven-module" );
-        op.setProjectTemplateName( "mvc-portlet" );
+        op.setProjectTemplateName( "npm-isomorphic-portlet" );
+
+        createAndBuild(op);
+    }
+
+    @Test
+    public void testProjectTemplateNpmJqueryPortlet() throws Exception
+    {
+        NewLiferayModuleProjectOp op = NewLiferayModuleProjectOp.TYPE.instantiate();
+
+        op.setProjectName( "npm-jquery-portlet-test" );
+        op.setProjectProvider( "maven-module" );
+        op.setProjectTemplateName( "npm-jquery-portlet" );
+
+        createAndBuild(op);
+    }
+
+    @Test
+    public void testProjectTemplateNpmMetaljsPortlet() throws Exception
+    {
+        NewLiferayModuleProjectOp op = NewLiferayModuleProjectOp.TYPE.instantiate();
+
+        op.setProjectName( "npm-metaljs-portlet-test" );
+        op.setProjectProvider( "maven-module" );
+        op.setProjectTemplateName( "npm-metaljs-portlet" );
+
+        createAndBuild(op);
+    }
+
+    @Test
+    public void testProjectTemplateNpmPortlet() throws Exception
+    {
+        NewLiferayModuleProjectOp op = NewLiferayModuleProjectOp.TYPE.instantiate();
+
+        op.setProjectName( "npm-portlet-test" );
+        op.setProjectProvider( "maven-module" );
+        op.setProjectTemplateName( "npm-portlet" );
+
+        createAndBuild(op);
+    }
+
+    @Test
+    public void testProjectTemplateNpmReactPortlet() throws Exception
+    {
+        NewLiferayModuleProjectOp op = NewLiferayModuleProjectOp.TYPE.instantiate();
+
+        op.setProjectName( "npm-react-portlet-test" );
+        op.setProjectProvider( "maven-module" );
+        op.setProjectTemplateName( "npm-react-portlet" );
+
+        createAndBuild(op);
+    }
+
+    @Test
+    public void testProjectTemplateNpmVuejsPortlet() throws Exception
+    {
+        NewLiferayModuleProjectOp op = NewLiferayModuleProjectOp.TYPE.instantiate();
+
+        op.setProjectName( "npm-vuejs-portlet-test" );
+        op.setProjectProvider( "maven-module" );
+        op.setProjectTemplateName( "npm-vuejs-portlet" );
 
```

Comment:
Do we have unit tests for all project types?

Reply:
For maven module, we have unit tests for all project types now. 
For gradle module, we don't have it. I will add it.